packages/teuchos/parameterlist/test/ParameterList/ParameterEntry_UnitTest.cpp

```diff
@@ -45,6 +45,47 @@
 
 namespace Teuchos {
 
+//! @test Check that the move constructor of @ref Teuchos::ParameterEntry works as expected.
+TEUCHOS_UNIT_TEST(Teuchos_ParameterEntry, move_constructor)
+{
+    ParameterEntry source(std::string("nice-entry"));
+    TEST_EQUALITY(Teuchos::any_cast<std::string>(source.getAny()),"nice-entry");
+
+    //! After the move-construct, @c source has no value (its holding a @c nullptr when looking at how @c Teuchos::any holds values).
+    ParameterEntry move_constructed(std::move(source));
+    TEST_EQUALITY(source.getAny().access_content(),nullptr);
+    TEST_EQUALITY(source.getAny().has_value(),false);
+    TEST_EQUALITY(Teuchos::any_cast<std::string>(move_constructed.getAny()),"nice-entry");
+
+    //! Same behavior is expected for the move-assign.
+    ParameterEntry move_assigned;
+    move_assigned = std::move(move_constructed);
+    TEST_EQUALITY(move_constructed.getAny().access_content(),nullptr);
+    TEST_EQUALITY(move_constructed.getAny().has_value(),false);
+    TEST_EQUALITY(Teuchos::any_cast<std::string>(move_assigned.getAny()),"nice-entry");
+}
+
+//! @test This test checks that it is possible to move-extract the value held by a parameter entry.
+TEUCHOS_UNIT_TEST(Teuchos_ParameterList, move_extract_value_from_any)
+{
+    ParameterEntry source(std::string("nice-entry"));
+
+    Teuchos::any::holder<std::string>* dyn_cast_content = dynamic_cast<Teuchos::any::holder<std::string>*>(
+        source.getAny().access_content()
+    );
```

Comment:
The style used in the rest of Teuchos is to line up the closing ')' with the arguments.  Also, Teuchos indents 2 spaces, not 4 spaces.

Reply:
I think I went over all the changes and tried to comply with:
- 2 spaces (instead of 4)
- clogin `)`